src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/RoslynCSharpTokenizer.cs

```diff
@@ -386,404 +210,472 @@ private StateResult Data()
-        }
-        else if (CurrentCharacter == '@')
-        {
-            // Could be escaped comment transition
-            return Transition(
-                CSharpTokenizerState.EscapedRazorCommentTransition,
-                EndToken(SyntaxKind.Transition));
+            case '"':
+                return TokenizedExpectedStringOrCharacterLiteral(CSharpSyntaxKind.StringLiteralToken, SyntaxKind.StringLiteral, expectedPrefix: "@\"", expectedPostfix: "\"");
+            case '$' when Peek(2) == '"':
+                return TokenizedExpectedStringOrCharacterLiteral(CSharpSyntaxKind.InterpolatedStringStartToken, SyntaxKind.StringLiteral, expectedPrefix: "@$\"", expectedPostfix: "\"");
+            case '*':
+                return Assumed.Unreachable<StateResult>();
+            case '@':
+                // Escaped razor transition. Likely will error in the parser.
+                AddResetPoint();
+                TakeCurrent();
+                _roslynTokenParser.SkipForwardTo(Source.Position);
+                AssertCurrent('@');
+                return Transition(RoslynCSharpTokenizerState.Token, EndToken(SyntaxKind.Transition));
+            default:
+                // Either a standard razor transition or a C# identifier. The parser will take care of stitching together the transition and the
+                // identifier if it's the latter case.
+                AddResetPoint();
+                TakeCurrent();
+                _roslynTokenParser.SkipForwardTo(Source.Position);
+                var trailingTrivia = GetNextResult(NextResultType.TrailingTrivia);
+                _currentCSharpTokenTriviaEnumerator = (trailingTrivia.Token.TrailingTrivia.GetEnumerator(), isLeading: false);
+                return Transition(RoslynCSharpTokenizerState.TriviaForCSharpToken, EndToken(SyntaxKind.Transition));
         }
-
-        return Stay(EndToken(SyntaxKind.Transition));
     }
 
-    private StateResult EscapedRazorCommentTransition()
+    private StateResult Operator()
     {
-        TakeCurrent();
-        return Transition(CSharpTokenizerState.Data, EndToken(SyntaxKind.Transition));
-    }
+        var curPosition = Source.Position;
+        var result = GetNextResult(NextResultType.Token);
+        var token = result.Token;
 
-    private SyntaxKind Operator()
-    {
-        var first = CurrentCharacter;
-        TakeCurrent();
-        Func<SyntaxKind> handler;
-        if (_operatorHandlers.TryGetValue(first, out handler))
-        {
-            return handler();
-        }
-        return SyntaxKind.Marker;
-    }
+        // Don't include trailing trivia; we handle that differently than Roslyn
+        var finalPosition = curPosition + token.Span.Length;
 
-    private SyntaxKind LessThanOperator()
-    {
-        if (CurrentCharacter == '=')
+        for (; curPosition < finalPosition; curPosition++)
         {
             TakeCurrent();
-            return SyntaxKind.LessThanEqual;
         }
```

Comment:
This code appears to be duplicated in multiple places in the class, could this be refactored into a private method that can be called from multiple call sites?